org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/LocalGradleDistribution.java

```diff
@@ -32,11 +31,23 @@
     private final File location;
 
     LocalGradleDistribution(File location) {
-        this.location = Preconditions.checkNotNull(location);
-        Optional<String> error = GradleDistributionInfo.validateLocalInstallationLocation(location);
+        this.location = Preconditions.checkNotNull(location).getAbsoluteFile();
+        Optional<String> error = validate(location);
         Preconditions.checkArgument(!error.isPresent(), error.or(""));
     }
 
+    private static  Optional<String> validate(File location) {
+        if (location == null) {
+            return Optional.of(NLS.bind(CoreMessages.ErrorMessage_0_MustBeSpecified, CoreMessages.GradleDistribution_Label_LocalInstallationDirectory));
```

Comment:
Just throw here instead of doing an Optional dance :)

Reply:
I removed the whole validate method.